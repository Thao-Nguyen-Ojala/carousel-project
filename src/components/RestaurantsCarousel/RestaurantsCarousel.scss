.carousel-container {
  padding-top: 1rem;
  width: 100%;

  &__wrapper {
    padding: 1rem 0;
  }  
}


.react-multiple-carousel__arrow--left {
  left: calc(1%) !important;
}

.react-multiple-carousel__arrow--right {
  right: calc(1%) !important;
}