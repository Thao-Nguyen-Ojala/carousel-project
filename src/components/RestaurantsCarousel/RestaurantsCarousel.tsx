import React from 'react';
import './RestaurantsCarousel.scss';
import { Header, VenueBanner } from '..';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import { RestaurantSection } from '../../restaurant.module';

const responsive = {
  desktop: {
    breakpoint: { max: 9999, min: 959 },
    items: 3,
  },
  tablet: {
    breakpoint: { max: 959, min: 700 },
    items: 2,
  },
  mobile: {
    breakpoint: { max: 700, min: 0 },
    items: 1,
  },
};

type RestaurantsCarouselPropsType = { restaurantSection: RestaurantSection };

const RestaurantsCarousel = ({
  restaurantSection,
}: RestaurantsCarouselPropsType) => {
  const { title, restaurants } = restaurantSection;
  return (
    <div className="carousel-container">
      <Header title={title} restaurantsCount={restaurants.length} />
      <div className="carousel-container__wrapper">
        <Carousel
          responsive={responsive}
          infinite={true}
          renderButtonGroupOutside={true}
          containerClass="carousel-container__showcase"
        >
          {restaurants.map((restaurant, i) => {
            return (
              <VenueBanner
                blurhash={restaurant.blurhash}
                name={restaurant.name}
                online={restaurant.online}
                key={i}
              />
            );
          })}
        </Carousel>
      </div>
    </div>
  );
};

export default RestaurantsCarousel;
