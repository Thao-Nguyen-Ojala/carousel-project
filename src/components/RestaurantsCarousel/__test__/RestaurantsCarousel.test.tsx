import React from 'react';
import ReactDOM from 'react-dom';
import RestaurantsCarousel from '../RestaurantsCarousel';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

const testingObj = {
  title: 'Popular Restaurants',
  restaurants: [
    {
      blurhash: 'UAN=8k?LS~M:ErJFs%t0MDMWRqo@%BxSV{RX',
      launch_date: '2020-04-20',
      location: [24.938082, 60.17626],
      name: 'Sea Chain',
      online: true,
      popularity: 0.956990414084132,
    },
  ],
};

const testingObjs2 = {
  title: 'Popular Restaurants',
  restaurants: [
    {
      blurhash: 'UAN=8k?LS~M:ErJFs%t0MDMWRqo@%BxSV{RX',
      launch_date: '2020-04-20',
      location: [24.938082, 60.17626],
      name: 'Sea Chain',
      online: true,
      popularity: 0.956990414084132,
    },
    {
      blurhash: 'UKNaZ$xnRXaQO5WEt2f7DfRpo?k8MptKV}ou',
      launch_date: '2020-03-14',
      location: [24.924752, 60.179213],
      name: 'Charming Pepper Emporium',
      online: true,
      popularity: 0.741748846018373,
    },
    {
      blurhash: 'ULFgVE}sE3bvGNORnUj=A1JCoaniMUj0tMS5',
      launch_date: '2020-09-24',
      location: [24.936465, 60.178633],
      name: 'Loving Meat Basket',
      online: true,
      popularity: 0.7400471016913404,
    },
    {
      blurhash: 'ULDK46wNI@jJ15NuxDbF0Ys+W.oc?8n+RoWF',
      launch_date: '2020-03-06',
      location: [24.943776, 60.153804],
      name: 'Vegetable Factory',
      online: true,
      popularity: 0.7332623605450638,
    },
    {
      blurhash: 'UQCBXU,@JQn,${xEWXWo31OTwzW-N4Ncs+sV',
      launch_date: '2020-01-12',
      location: [24.925688, 60.178389],
      name: 'Cucumber Hotel',
      online: true,
      popularity: 0.6259073873586668,
    },
  ],
};

it('render RestaurantsCarousel without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <RestaurantsCarousel restaurantSection={testingObj}></RestaurantsCarousel>,
    div
  );
});

it('render RestaurantsCarousel a restaurant array', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <RestaurantsCarousel
      restaurantSection={testingObjs2}
    ></RestaurantsCarousel>,
    div
  );
});

it('matches RestaurantsCarousel snapshot', () => {
  const testingRestaurantsCarousel = renderer
    .create(
      <RestaurantsCarousel restaurantSection={testingObj}></RestaurantsCarousel>
    )
    .toJSON();
  expect(testingRestaurantsCarousel).toMatchSnapshot();
});
