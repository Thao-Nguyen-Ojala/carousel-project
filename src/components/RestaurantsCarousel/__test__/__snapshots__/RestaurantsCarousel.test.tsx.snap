// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches RestaurantsCarousel snapshot 1`] = `
<div
  className="carousel-container"
>
  <div
    className="header-wrapper"
  >
    <span>
      <div
        className="header-wrapper__title"
      >
        Popular Restaurants
      </div>
      <div
        className="header-wrapper__restaurants-count"
      >
        (1 restaurants)
      </div>
    </span>
  </div>
  <div
    className="carousel-container__wrapper"
  >
    <div
      className="react-multi-carousel-list carousel-container__showcase "
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
